### PR TITLE
[YUNIKORN-1323] Add e2e tests for group handling

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
End-to-end tests which validate that the user-info annotation is properly added by the admission controller.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1323

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: test/e2e/admission_controller/admission_controller_suite_test.go [hunk 1]
```diff
@@ -58,6 +60,39 @@ var testPod = v1.Pod{
...
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "sleepjob",
+						Image:   "alpine:latest",
+						Command: []string{"sleep", "30"},
+					},
+				},
+			},
+		},
```
Comment (wilfred-s) [inline test/e2e/admission_controller/admission_controller_suite_test.go:86]: This template will come back in a number of places we might want to factor this out so we can re-use.

**Review by wilfred-s — COMMENTED** [review]
Is it possible to check the "do not trust controllers" case inside the e2e tests?
Would be nice to have that one added if possible.
Checking user allowed users is I think not possible with the test setup we use which is OK to skip.

File: test/e2e/admission_controller/admission_controller_suite_test.go (hunk 1 above)
Comment (pbacsko) [reply test/e2e/admission_controller/admission_controller_suite_test.go]: Done

**Comment by pbacsko** [comment]
> Is it possible to check the "do not trust controllers" case inside the e2e tests? Would be nice to have that one added if possible. Checking user allowed users is I think not possible with the test setup we use which is OK to skip.

Actually both can be tested, because the configmap can be updated and changes are applied on-the-fly.

**Comment by wilfred-s** [comment]
Kicked off the two failed tests again. K8s 1.25 failed for both default and plugin but at different points.
Does not look like a code issue.

**Review by wilfred-s — APPROVED** [review]
LGTM, would like to see the K8s 1.25 tests pass before we commit

**Comment by pbacsko** [comment]
> LGTM, would like to see the K8s 1.25 tests pass before we commit

We have a green build, but I think the tests are still flaky. We can give it a go and if they're very unstable then we'll figure out something.

File: test/e2e/admission_controller/admission_controller_test.go [hunk 2]
```diff
@@ -107,6 +110,139 @@ var _ = ginkgo.Describe("AdmissionController", func() {
...
+		gomega.Ω(deployment.Status.ReadyReplicas).To(gomega.Equal(int32(0)))
+
+		// restore setting
+		ginkgo.By("Restore trustController setting")
+		configMap, err = kubeClient.GetConfigMap(constants.ConfigMapName, configmanager.YuniKornTestConfig.YkNamespace)
+		gomega.Ω(err).ShouldNot(gomega.HaveOccurred())
+		configMap.Data[amConf.AMAccessControlTrustControllers] = "true"
+		_, err = kubeClient.UpdateConfigMap(configMap, configmanager.YuniKornTestConfig.YkNamespace)
+		gomega.Ω(err).ShouldNot(gomega.HaveOccurred())
+
```
Comment (wilfred-s) [inline test/e2e/admission_controller/admission_controller_test.go:194]: Should have a sleep here like in the other places we do the update

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: test/e2e/admission_controller/admission_controller_test.go (hunk 2 above)
Comment (craigcondit) [reply test/e2e/admission_controller/admission_controller_test.go:229]: That will definitely introduce a delay, but doesn't verify that YuniKorn has applied the config. We may want to follow-up later with a more comprehensive method to check that config has been applied. This is fine for now though.

**Review by wilfred-s — APPROVED** [review]
LGTM
the eventhandler waits for a notify that of the change, ew then give the system 1 second to process the change that should be more than enough

**Pull request closed without merging**